[Catalyst][Feature] GitHub Integration (#55)
Xerilium/catalyst: src/templates/specs/plan.md

## Project Structure

> [INSTRUCTIONS]
> Show folder architecture and primary entry point files this feature creates. Focus on structure, not exhaustive file lists.
>
> **When to list individual files:**
> - Feature has ≤10 key files AND file organization illustrates a specific architectural pattern (e.g., adapter pattern, plugin architecture)
> - Otherwise, show folders only with descriptions of what goes in them
>

> > **Comments:** Only add inline comments when clarifying non-obvious purpose. Omit comments for self-documenting filenames (types.ts, errors.ts, models/, tests/).

Author: ```suggestion
> **Comments:** Only add inline comments to clarify non-obvious names. Don't use comments for self-documenting names.
```
